pkg/chartmuseum/router/match_test.go

```diff
@@ -0,0 +1,182 @@
+			if route != nil {
+				route.Handler(c)
+			}
+			val, exists = c.Get("index")
+			suite.True(exists)
+			suite.Equal(2, val)
+			suite.Equal([]gin.Param{{"repo", repo}}, params)
+
+			// GET /charts/mychart-0.1.0.tgz
+			r = pathutil.Join("/", contextPath, repo, "charts/mychart-0.1.0.tgz")
+			route, params = match(routes, "GET", r, contextPath, depth)
+			suite.NotNil(route)
+			if route != nil {
+				route.Handler(c)
+			}
+			val, exists = c.Get("index")
+			suite.True(exists)
+			suite.Equal(3, val)
+			suite.Equal([]gin.Param{{"filename", "mychart-0.1.0.tgz"}, {"repo", repo}}, params)
+
+			// GET /api/charts
+			r = pathutil.Join("/", contextPath, "api", repo, "charts")
+			route, params = match(routes, "GET", r, contextPath, depth)
+			suite.NotNil(route)
+			if route != nil {
+				route.Handler(c)
+			}
+			val, exists = c.Get("index")
+			suite.True(exists)
+			suite.Equal(4, val)
+			suite.Equal([]gin.Param{{"repo", repo}}, params)
+
+			// GET /api/charts/mychart
+			r = pathutil.Join("/", contextPath, "api", repo, "charts/mychart")
+			route, params = match(routes, "GET", r, contextPath, depth)
+			suite.NotNil(route)
+			if route != nil {
+				route.Handler(c)
+			}
+			val, exists = c.Get("index")
+			suite.True(exists)
+			suite.Equal(5, val)
+			suite.Equal([]gin.Param{{"name", "mychart"}, {"repo", repo}}, params)
+
+			// GET /api/charts/mychart/0.1.0
+			r = pathutil.Join("/", contextPath, "api", repo, "charts/mychart/0.1.0")
+			route, params = match(routes, "GET", r, contextPath, depth)
+			suite.NotNil(route)
+			if route != nil {
+				route.Handler(c)
+			}
+			val, exists = c.Get("index")
+			suite.True(exists)
+			suite.Equal(6, val)
+			suite.Equal([]gin.Param{{"name", "mychart"}, {"version", "0.1.0"}, {"repo", repo}}, params)
+
+			// POST /api/charts
+			r = pathutil.Join("/", contextPath, "api", repo, "charts")
+			route, params = match(routes, "POST", r, contextPath, depth)
+			suite.NotNil(route)
+			if route != nil {
+				route.Handler(c)
+			}
+			val, exists = c.Get("index")
+			suite.True(exists)
+			suite.Equal(7, val)
+			suite.Equal([]gin.Param{{"repo", repo}}, params)
+
+			// POST /api/prov
+			r = pathutil.Join("/", contextPath, "api", repo, "prov")
+			route, params = match(routes, "POST", r, contextPath, depth)
+			suite.NotNil(route)
+			if route != nil {
+				route.Handler(c)
+			}
+			val, exists = c.Get("index")
+			suite.True(exists)
+			suite.Equal(8, val)
+			suite.Equal([]gin.Param{{"repo", repo}}, params)
+
+			// DELETE /api/charts/mychart/0.1.0
+			r = pathutil.Join("/", contextPath, "api", repo, "charts/mychart/0.1.0")
+			route, params = match(routes, "DELETE", r, contextPath, depth)
+			suite.NotNil(route)
+			if route != nil {
+				route.Handler(c)
+			}
+			val, exists = c.Get("index")
+			suite.True(exists)
+			suite.Equal(9, val)
+			suite.Equal([]gin.Param{{"name", "mychart"}, {"version", "0.1.0"}, {"repo", repo}}, params)
+		}
+	}
+}
+
+func TestMatchTestSuite(t *testing.T) {
```

Comment:
http://cs-guy.com/blog/2015/01/test-main/

Reply:
Test Main is more idiomatic but it's not a big deal ^^

Reply:
Ah. I am just following the convention in other test.go files. Lets update in near future for all.